src/torchmetrics/detection/helpers.py

```diff
@@ -89,10 +89,10 @@ def _fix_empty_tensors(boxes: Tensor) -> Tensor:
 
 
 def _validate_iou_type_arg(
-    iou_type: Union[Literal["bbox", "segm"], tuple[str]] = "bbox",
+    iou_type: Union[Literal["bbox", "segm", "keypoints"], tuple[str]] = "bbox",
 ) -> tuple[str]:
     """Validate that iou type argument is correct."""
-    allowed_iou_types = ("segm", "bbox")
+    allowed_iou_types = ("segm", "bbox", "keypoints")
```

Comment:
if we do not allow `"keypoints"` to be used together with the other iou types, we should validate for that in this function